src/components/QuranReader/index.tsx

```diff
@@ -55,47 +60,60 @@ const QuranReader = ({ initialData, chapter }: QuranReaderProps) => {
     },
     verseFetcher,
     {
-      initialData: initialData.verses,
+      initialData: isUsingDefaultTranslations ? initialData.verses : null, // initialData is set to null if the user changes/has changed the default translations so that we can prevent the UI from falling back to the default translations while fetching the verses with the translations the user had selected and we will show a loading indicator instead.
       revalidateOnFocus: false, // disable auto revalidation when window gets focused
       revalidateOnMount: true, // enable automatic revalidation when component is mounted. This is needed when the translations inside initialData don't match with the user preferences and would result in inconsistency either when we first load the QuranReader with pre-saved translations from the persistent store or when we change the translations' preferences after initial load.
     },
   );
   const readingView = useSelector(selectReadingView);
-  const pageLimit = isVerseView ? 1 : initialData.pagination.totalPages;
-  const verses = data.flat(1);
+  let body;
   let view;
-
-  if (readingView === ReadingView.QuranPage) {
-    view = <PageView verses={verses} />;
+  // if we are fetching the data (this will only happen when the user has changed the default translations so the initialData will be set to null).
+  if (!data) {
+    // TODO: add a proper loading indicator
+    body = <StyledLoading>loading...</StyledLoading>;
```

Comment:
🤔

The api loads pretty fast, so it's not a concern for now. 

Thinking out loud: the loading indicator currently blocks the entire container. Ideally, the loading indicator only blocks the translations component. 

When a user switches their translation, a good experience would be the following: 

1) the arabic verse is still visible
2) they pick up where they last left off, if they are reading verse 100, they'd still be on verse 100.

Probably going to be more work than the scope of this PR, but FYI.

Reply:
Okay got it. InshAllah in the future we can fine-tune this to only show the loading for the translations instead of the Quran text itself.